# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/GlobalObject.cpp
# Commit: cab0e02d59e1
# Full Hash: cab0e02d59e1b9b8ebff156bb63128aa4e8d8c5c
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-09-15 15:26:38
# Regressor Bug: 1730699
# File Overlap Count: 1
# Description:
#   Bug 1730699 part 2 - Cache empty iterator on the global. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D125552
# ==============================================================================

diff -r c7eae2ddd714 -r cab0e02d59e1 js/src/vm/GlobalObject.cpp
--- a/js/src/vm/GlobalObject.cpp	Wed Sep 15 09:20:44 2021 +0000
+++ b/js/src/vm/GlobalObject.cpp	Wed Sep 15 09:20:45 2021 +0000
@@ -1194,6 +1194,7 @@
   TraceNullableEdge(trc, &realmKeyObject, "global-realm-key");
   TraceNullableEdge(trc, &throwTypeError, "global-throw-type-error");
   TraceNullableEdge(trc, &eval, "global-eval");
+  TraceNullableEdge(trc, &emptyIterator, "global-empty-iterator");
 
   TraceNullableEdge(trc, &arrayShapeWithDefaultProto, "global-array-shape");
 